nvtabular/ops/categorify.py

```diff
@@ -339,6 +349,15 @@ def fit(self, col_selector: ColumnSelector, ddf: dd.DataFrame):
             if c not in cols_with_vocabs
         ]
 
+        # Define a rough row-count at which we are likely to
+        # start hitting memory-pressure issues that cannot
+        # be accommodated with smaller partition sizes.
+        # By default, we estimate a "problematic" cardinality
+        # to be one that consumes >12.5% of the total memory.
+        self.cardinality_memory_limit = parse_bytes(
+            self.cardinality_memory_limit or int(device_mem_size(kind="total", cpu=_cpu) * 0.125)
```

Comment:
Does this 12.5% relate to the 10% above?  Or is this specific to CPU memory?

Reply:
Oops. These should both be 12.5% (just pushed a correction).

Also, this is compared to 12.5% of the total host memory where the worker is running, or 12.5% of the total device memory for a single GPU, depending on whether we are running in cpu or gpu mode.

I am actually pretty unsatisfied with this answer, because this means the comparison is quite different for gpu and cpu-backed data. However, I am not sure if it is possible for us to query to query the total host memory allocated to the current Dask worker. We may be able to "fix" this in `device_mem_size` by searching for a global Dask client, but we probably need to make some of the changes suggested in #1249 first.